### DISCUSSION TITLE
For a Flutter app with companion native iOS and watchOS, does the workflow editor allow enabling both Flutter and XC/TestPlans?

### DISCUSSION THREAD

**Discussion by importRyan** [codemagic-ci-cd/codemagic-docs#2628] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

1. Open web workflow editor for an existing Flutter project
2. Tap Tests > Integration and unit tests

### Expected results

Option to detect/specify and run Test Plans for XCTest/Swift Testing.

### Actual results

Only options for Flutter tests

**Comment by himesh-cm** [comment]
Hey @importRyan, 

Running Flutter tests and XC tests for watchOS are different processes. Flutter commands (flutter test, flutter drive) are specifically for running tests written in Dart for your Flutter application. The `yaml` approach will give you more customisation options. 

But For your XC tests in the Flutter workflow editor, you can try and see if are okay with the approach. You can do it by specifying the commands in your `post-test / build script` as shown below.

![Screen Shot 2024-07-08 at 14 30 57 PM](https://github.com/codemagic-ci-cd/codemagic-docs/assets/106069365/1833f4c7-ed8f-446b-9909-23c93367f04d)

Let me know if that answers your question or if you have any further queries.

**Comment by himesh-cm** [comment]
I'm tentatively closing the discussion since we haven't heard back from you. You can open the discussion again by replying to this message or opening a new discussion.
